### PR TITLE
feat(ui): widen the agent chat API surface for embedding consumers

### PR DESCRIPTION
**Pull request by cmorman89**
## Summary

Consumers that embed `ModernAgentConversation` currently have to copy logic out of `@vertesia/ui` to build anything alongside it. Two separate causes:

1. **Reachability.** A symbol having `export` at the file level is not enough. Anything resolving `@vertesia/ui/*` against the package's `exports` map can only reach what a *published subpath's barrel chain* re-exports. `features/agent/chat/index.ts` uses an explicit component list and never surfaced `./hooks` or `./workstreams`, so those symbols were invisible despite being exported.
2. **Module privacy.** Several pieces of reusable logic were module-private inside `ModernAgentConversation.tsx`.

This PR widens the API surface. No behaviour change for existing consumers — it only makes more things importable.

### Surfaced through `@vertesia/ui/features` (barrel only, no logic change)

- `useDocumentPanel` + `UseDocumentPanelResult`
- `useArtifacts` + `UseArtifactsResult` + `ArtifactTreeNode`
- `export * from './workstreams.js'` — `formatWorkstreamName`, `getWorkstreamStatusClass`, `getWorkstreamLaunchDetails`, the lifecycle/internal-result predicates, and the `WorkstreamInfo` type

### Extracted from `ModernAgentConversation.tsx`

- **`chat/deriveWorkstreams.ts`** (new) — `deriveWorkstreamsFromMessages` plus the `getActiveWorkstreams` merge helpers (`mergeWorkstreamInfo`, `mergePreservingTerminalStatus`, `activeWorkstreamEntryToInfo`, `completedWorkstreamEntryToInfo`, `isActiveWorkstreamStatus`, `isTerminalWorkstreamStatus`).

  A sibling module rather than folding into `workstreams.ts`: `ModernAgentOutput/utils.ts` already imports from `workstreams.ts`, and the derivation needs `getWorkstreamId` from utils — merging them would introduce an import cycle. This way no existing consumer changes.

- **`deriveArtifactRefreshKey(messages)`** → exported from `chat/hooks/useArtifacts.ts`, colocated with the hook it feeds.

- **`useActiveWorkstreams(client, runId, enabled, pollIntervalMs?)`** → new `chat/hooks/useActiveWorkstreams.ts`. The hook owns the poll interval, the unavailability latch, the warn-once ref and the reset-on-`runId`; callers pass only the conditions they own.

### Backported from `main`

`OverflowTabsBar` / `OverflowMoreMenu` / `useOverflowTabs` from #1896, so the measure-and-collapse tab bar is reusable from `@vertesia/ui/core` rather than living inside `AgentRightPanel`. `WorkstreamTabs.tsx` ends up byte-identical to its post-#1896 state on `main`.

## Behaviour

The moved logic is byte-identical to the originals — verified by diffing the 221 moved executable lines against the pre-change source. One intentional difference: the old polling effect re-ran on any `initialHistoryStatus` change, so a transition between two non-`loading` values (e.g. `empty` → `loaded`) restarted the poll and fired a redundant immediate fetch. `enabled` collapses those conditions, so that no longer happens.

Related: `debugAgentChat('active workstreams poll state', …)` now logs a collapsed `enabled` boolean instead of the three source conditions — the hook cannot see the caller's sub-conditions.

## Test Plan

- [x] `turbo build` — 56/56 tasks pass (full workspace, incl. all dependents of `@vertesia/ui`)
- [x] `pnpm test` in `packages/ui` — 454 tests / 41 files pass, including `ModernAgentConversation.test.tsx` and `AgentChatFixtureReplay.test.tsx`
- [x] `pnpm run lint` — clean (1130 files)
- [x] `pnpm run format:check` — clean (1130 files)
- [x] **Reachability check** — extracted the export lists from the built rollup bundles and confirmed all 21 new symbols resolve from `@vertesia/ui/features` and all 4 from `@vertesia/ui/core`. This is the check that matters: a symbol can compile fine and still fail to resolve at runtime for a consumer that externalizes `@vertesia/ui/*`.
- [x] Type surface — `lib/features/agent/chat/index.d.ts` re-exports all new names; `.d.ts` emitted for both new modules

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01F2PTudPBFXoVWka1pPqbgj

### CODE REVIEW

**Comment by michaelva** [comment]
Thanks for splitting this into a narrow export change. I verified the barrel chain and do not see an export collision around `WorkstreamInfo`. Before exposing these hooks as public API, I found three lifecycle/API gaps:

1. **`useDocumentPanel` needs a conversation identity/reset key.** It only resets when `messages.length === 0`. If a consumer switches directly from one non-empty history to another, `lastProcessedIndex` and `openDocuments` belong to the previous run. The new history can be skipped entirely or combined with stale documents. Please accept a `runId`/conversation key and reset `lastProcessedIndex`, documents, selection, panel state, and refresh state whenever it changes.

2. **`useArtifacts` should clear state when `runId` becomes undefined.** The current early return leaves the previous run files/error visible indefinitely. The no-run path should invalidate any pending fetch and clear `allFiles`, `error`, and loading state.

3. **Export `OpenDocument`.** `UseDocumentPanelResult.openDocuments` exposes `OpenDocument[]`, but the canonical type is not reachable from the public barrel. Consumers can derive it through indexed access, but cannot import the type directly. Please add:
```ts
export type { OpenDocument } from './types/document.js';
```

Non-blocking API hygiene: an explicit workstream export list would better define the supported contract than `export *`; the star currently publishes internal-result helpers and will automatically publish future additions.

Merge coordination: this PR and #1907 edit the same location in `chat/index.ts` and conflict textually. After one lands, the other needs a rebase retaining both export blocks.

Repository policy also requires removing the Claude co-author/session trailers from the commit before merge.

File: packages/ui/src/features/agent/chat/index.ts [hunk 1]
```diff
@@ -86,3 +95,5 @@ export {
 export * from './SkillWidgetProvider';
 export { VegaLiteChart } from './VegaLiteChart';
 export * from './WaitingMessages';
+// Workstream naming / status / lifecycle helpers and the WorkstreamInfo shape
+export * from './workstreams.js';
```
Comment (michaelva) [inline packages/ui/src/features/agent/chat/index.ts]: Non-blocking API-contract suggestion: prefer an explicit export list here. `export *` currently publishes helpers named as internals and will silently expand the supported public surface whenever a future implementation helper is exported from `workstreams.ts`. An explicit list makes additions to this public contract reviewable.

File: packages/ui/src/features/agent/chat/index.ts [hunk 2]
```diff
@@ -37,6 +37,15 @@ export {
     ConfirmationWidget,
     type ConfirmationWidgetProps,
 } from './AskUserWidget';
+// Artifact listing hook (run artifacts tree) for external use
+export {
+    type ArtifactTreeNode,
+    isSystemArtifact,
+    type UseArtifactsResult,
+    useArtifacts,
+} from './hooks/useArtifacts.js';
```
Comment (michaelva) [inline packages/ui/src/features/agent/chat/index.ts:46]: Before making `useArtifacts` public, please handle the no-run transition in the hook. Its current `if (!runId) return` leaves files and errors from the previous run visible indefinitely when an embedding app resets to its start view. The no-run path should increment `fetchIdRef` to invalidate pending requests, then clear `allFiles`, `error`, and `isLoading`. A rerender test from `runId="run-a"` to `undefined` would guard this contract.

File: packages/ui/src/features/agent/chat/index.ts [hunk 3]
```diff
@@ -37,6 +37,15 @@ export {
...
+// Document side-panel hook for external use
+export { type UseDocumentPanelResult, useDocumentPanel } from './hooks/useDocumentPanel.js';
```
Comment (michaelva) [inline packages/ui/src/features/agent/chat/index.ts:48]: This public export needs two related fixes:

1. Give `useDocumentPanel` a conversation identity/reset key. It currently resets only when `messages.length === 0`; switching directly between non-empty histories retains `lastProcessedIndex` and documents from the previous run, so the new history can be skipped or mixed with stale state. Reset the ref, documents, selection, panel state, and refresh key when the identity changes.
2. Re-export the canonical item type used by `UseDocumentPanelResult.openDocuments`:
```ts
export type { OpenDocument } from './types/document.js';
```

Please add a rerender test that switches directly from one non-empty conversation to another.

**Comment by cmorman89** [comment]
Thanks — all three were correct, and I verified each against the code before acting. Pushed `f87b2fb3`.

**`useArtifacts` no-run transition.** Confirmed: `if (!runId) return` sat *above* the `fetchIdRef` bump, so `allFiles`, `error` and `isLoading` all kept the previous run's values, and an in-flight listing could still land afterwards. The no-run path now bumps `fetchIdRef` first, then clears — with `setAllFiles((prev) => prev.length === 0 ? prev : [])` so an already-empty hook doesn't re-render.

**`useDocumentPanel` identity.** Confirmed, and worse than "skipped or mixed" implies in one direction: A(10 msgs) → B(8 msgs) leaves `lastProcessedIndex` at 9, so `startIdx = 10 >= 8` returns immediately and B is skipped *entirely* while A's documents stay open. Added an optional `conversationId`; the reset effect is declared before the scanning effect so the cursor is rewound in the same commit the new messages arrive. `ModernAgentConversation` now passes `agentRunId`.

**`export *`.** Agreed, and it was publishing 11 symbols. Replaced with an explicit list.

**`OpenDocument`.** Added. Worth noting this one is squarely my miss: I had reasoned through exactly this failure mode for `WorkstreamInfo` elsewhere in this workstream and never generalised it into a check over every exported signature.

**Tests.** Both rerender tests you asked for are in, plus stale-response and control cases — 7 new assertions across `useArtifacts.test.tsx` and `useDocumentPanel.test.tsx`. I verified they *fail* against the previous behaviour rather than just pass against the new: reverting the two hooks fails exactly 4 of the 7 (the three controls pass either way, as they should).

Full workspace build 60/60, 758 tests / 82 files pass, lint and format clean.
